### ISSUE TITLE
How to convert a ty to a string

### ISSUE THREAD

**Issue by JelteF** [dtolnay/syn#73]
Is there a way to get similar behaviour as from `syntax::print::pprust::ty_to_string`. I need to make a hashmap of `Ty`'s. Before macros 1.1 I used `syntax::print::pprust::ty_to_string` to make a hashable `Ty`.

**Comment by dtolnay** [comment]
This should do the trick: `quote!(#ty).to_string()`. Would it be better if I add `#[derive(Hash)]` to all the syn types?

**Comment by JelteF** [comment]
That would be even better, as long as the same Ty from different places in
the AST would be equal.

**Comment by dtolnay** [comment]
In 9bf4af8b5d39cff5bc9db66e09803efe60c2d16c I added Hash implementations for all the types and published it as syn 0.10.6. I hope this helps! The same Ty from different places in the AST will have the same hash.

**Comment by JelteF** [comment]
Thanks, that will make my code much nicer! I'll let you know if I run into
any more problems. So far I enjoy this lib way more than the rustc
internals I used before :), great library!

On Sat, 7 Jan 2017, 20:33 David Tolnay, <notifications@github.com> wrote:

In 9bf4af8
<https://github.com/dtolnay/syn/commit/9bf4af8b5d39cff5bc9db66e09803efe60c2d16c>
I added Hash implementations for all the types and published it as syn
0.10.6. I hope this helps! The same Ty from different places in the AST
will have the same hash.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/dtolnay/syn/issues/73#issuecomment-271104791>, or mute
the thread
